plonk-core/src/circuit.rs

```diff
@@ -605,12 +595,29 @@ mod test {
     #[test]
     #[allow(non_snake_case)]
     fn test_full_on_Bls12_381() -> Result<(), Error> {
-        test_full::<Bls12_381, ark_ed_on_bls12_381::EdwardsParameters>()
+        test_full::<
+            <Bls12_381 as PairingEngine>::Fr,
+            ark_ed_on_bls12_381::EdwardsParameters,
+            crate::commitment::KZG10<Bls12_381>,
+        >()?;
+
+        test_full::<
```

Comment:
I think it's best to separate in 2 tests :
`test_full_on_Bls12_381` and `test_full_on_Bls12_381_ipa`